Initial schema creation fails with Oracle 18c XE

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

<!-- Explain the bug, if an error is being thrown a stack trace helps -->
Initial schema creation fails when the Oracle DB (at least for 18c XE) is used.

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [com/mycompany/myapp/config/LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/00000000000000_initial_schema.xml::00000000000001::jhipster:
     Reason: java.lang.NullPointerException
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1778)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:593)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:307)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:303)
	... 98 common frames omitted
Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/00000000000000_initial_schema.xml::00000000000001::jhipster:
     Reason: java.lang.NullPointerException
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:637)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:53)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:83)
	at liquibase.Liquibase.update(Liquibase.java:202)
	at liquibase.Liquibase.update(Liquibase.java:179)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:353)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:305)
	at org.springframework.boot.autoconfigure.liquibase.DataSourceClosingSpringLiquibase.afterPropertiesSet(DataSourceClosingSpringLiquibase.java:46)
	at io.github.jhipster.config.liquibase.AsyncSpringLiquibase.initDb(AsyncSpringLiquibase.java:118)
	at io.github.jhipster.config.liquibase.AsyncSpringLiquibase.afterPropertiesSet(AsyncSpringLiquibase.java:103)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1837)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1774)
	... 107 common frames omitted
Caused by: java.lang.NullPointerException: null
	at liquibase.change.core.LoadDataChange.retrieveMissingColumnLoadTypes(LoadDataChange.java:628)
	at liquibase.change.core.LoadDataChange.generateStatements(LoadDataChange.java:283)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1209)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:600)
	... 118 common frames omitted
```

##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->
A fresh app must starts normally.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->
- Generate a fresh new app selecting Oracle
- Start the app and it will fail

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->
Add column definition inside `loadData` tag.
Fix from sample app https://github.com/pmverma/jhipster-hypersistence-oracle/commit/2b2dafe5bc5451f7503e7757ed17c9f9940edf6d
This will fix for Oracle, however, I have not tested for other DBs by adding column definition.

##### **JHipster Version(s)**

```
jhipster@0.0.0 /Users/sierra/github/temp/jh-ho-oracle
└── generator-jhipster@6.4.1 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": true,
    "enableSwaggerCodegen": false,
    "enableTranslation": false,
    "jhiPrefix": "jhi",
    "languages": [
      "en",
      "fr"
    ],
    "messageBroker": false,
    "nativeLanguage": "en",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "prodDatabaseType": "oracle",
    "searchEngine": false,
    "serviceDiscoveryType": false,
    "skipClient": true,
    "skipServer": false,
    "testFrameworks": [],
    "websocket": false,
    "applicationType": "monolith",
    "baseName": "jhipster",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "buildTool": "maven",
    "useSass": true,
    "jhipsterVersion": "6.4.1",
    "skipUserManagement": false,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "embeddableLaunchScript": false,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "blueprints": []
  },
  "entities": [
    "EFive",
    "EFour",
    "EOne",
    "ESix",
    "EThree",
    "ETwo"
  ]
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity EFive {
  five String required
}
entity EFour {
  four AnyBlob required
}
entity EOne {
  one String required unique
}
entity ESix {
  six String required
}
entity EThree {
  three Instant required
}
entity ETwo {
  two Integer required
}
relationship OneToOne {
  ESix{user} to User with useJPADerivedIdentifier,
  EThree{user} to User
}
relationship OneToMany {
  EOne{eTwoO required} to ETwo{eOne required}
}
relationship ManyToOne {
  ETwo{eTwoT} to EOne
}
relationship ManyToMany {
  EFour{eFive} to EFive{eFour}
}

paginate EFive with pagination
paginate EOne with infinite-scroll
paginate ETwo with pager
service EFive, EFour, EOne, ESix, EThree, ETwo with serviceImpl
dto EThree with mapstruct

</pre>
</details>


##### **Environment and Tools**

openjdk version "1.8.0_222"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_222-b10)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.222-b10, mixed mode)

git version 2.20.1 (Apple Git-117)

node: v10.14.2

npm: 6.11.2

yeoman: 3.1.0

Docker version 19.03.2, build 6a30dfc

docker-compose version 1.24.1, build 4667896b

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

-   [ ] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
+++ b/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
@@ -163,7 +163,11 @@
                   <%_ if (devDatabaseType === 'mssql' || prodDatabaseType === 'mssql') { _%>
                   identityInsertEnabled="true"
                   <%_ } _%>
-                  tableName="<%= jhiTablePrefix %>_authority"/>
+                  tableName="<%= jhiTablePrefix %>_authority">
+            <%_ if (devDatabaseType === 'oracle') { _%>
+            <column name="name" type="string"/>
+            <%_ } _%>
+        </loadData>
 
         <<% if (devDatabaseType === 'mssql' || prodDatabaseType === 'mssql') { %>ext:<% } %>loadData
                   file="config/liquibase/data/user_authority.csv"
@@ -171,7 +175,12 @@
                   <%_ if (devDatabaseType === 'mssql' || prodDatabaseType === 'mssql') { _%>
                   identityInsertEnabled="true"
                   <%_ } _%>
-                  tableName="<%= jhiTablePrefix %>_user_authority"/>
+                  tableName="<%= jhiTablePrefix %>_user_authority">
+            <%_ if (devDatabaseType === 'oracle') { _%>
+            <column name="user_id" type="numeric"/>
+            <column name="name" type="string"/>
+            <%_ } _%>
+        </loadData>
     <%_ } _%>
 <%_ } _%>
         <createTable tableName="<%= jhiTablePrefix %>_persistent_audit_event">

--- a/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
+++ b/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
@@ -164,9 +164,7 @@
                   identityInsertEnabled="true"
                   <%_ } _%>
                   tableName="<%= jhiTablePrefix %>_authority">
-            <%_ if (devDatabaseType === 'oracle' || prodDatabaseType == 'oracle') { _%>
             <column name="name" type="string"/>
-            <%_ } _%>
         </loadData>
 
         <<% if (devDatabaseType === 'mssql' || prodDatabaseType === 'mssql') { %>ext:<% } %>loadData
@@ -176,10 +174,7 @@
                   identityInsertEnabled="true"
                   <%_ } _%>
                   tableName="<%= jhiTablePrefix %>_user_authority">
-            <%_ if (devDatabaseType === 'oracle' || prodDatabaseType === 'oracle') { _%>
             <column name="user_id" type="numeric"/>
-            <column name="name" type="string"/>
-            <%_ } _%>
         </loadData>
     <%_ } _%>
 <%_ } _%>

--- a/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
+++ b/generators/server/templates/src/main/resources/config/liquibase/changelog/initial_schema.xml.ejs
@@ -164,7 +164,7 @@
                   identityInsertEnabled="true"
                   <%_ } _%>
                   tableName="<%= jhiTablePrefix %>_authority">
-            <%_ if (devDatabaseType === 'oracle') { _%>
+            <%_ if (devDatabaseType === 'oracle' || prodDatabaseType == 'oracle') { _%>
             <column name="name" type="string"/>
             <%_ } _%>
         </loadData>
@@ -176,7 +176,7 @@
                   identityInsertEnabled="true"
                   <%_ } _%>
                   tableName="<%= jhiTablePrefix %>_user_authority">
-            <%_ if (devDatabaseType === 'oracle') { _%>
+            <%_ if (devDatabaseType === 'oracle' || prodDatabaseType === 'oracle') { _%>
             <column name="user_id" type="numeric"/>
             <column name="name" type="string"/>
             <%_ } _%>

